src/components/Viewer/Viewer.tsx

```diff
@@ -51,45 +85,62 @@ const Viewer: React.FC<ViewerProps> = ({ manifest, theme }) => {
       );
       setPainting({ ...painting });
     }
+    setResources(resources);
+    resources.length === 0 ? setIsNavigator(false) : setIsNavigator(true);
```

Comment:
Same here.... `setIsNavigator(resources.length === 0)`

Reply:
Yea, anytime you see 2 identical variables or expressions, there's generally a trick to tighten 'em up.  Less code, easier to read in a few months.